### PR TITLE
Only copy credentials for the cloud used to deploy OCP

### PR DESCRIPTION
**Pull request by flaper87**
We should copy only the cloud enabled by `OS_CLOUD`, rather than copying the entire clouds.yaml available in the node that runs `openshift-install`. 

In addition, this PR switches to use `go-yaml` rather than `ghodss` as the former does not serialize the map keys properly. We should use `go-yaml`, which is the same used by `clientconfig`

/cc @hardys @russellb

### CODE REVIEW

**Comment by crawford** [comment]
/hold

We explicitly switched from go-yaml before (cc @abhinavdahiya). We are currently trying to get a critical set of patches in.

**Comment by abhinavdahiya** [comment]
> In addition, this PR switches to use go-yaml rather than ghodss as the former does not serialize the map keys properly. We should use go-yaml, which is the same used by clientconfig

Any examples?

more info on why k8s uses ghodss https://github.com/kubernetes-sigs/yaml

**Comment by flaper87** [comment]
> > In addition, this PR switches to use go-yaml rather than ghodss as the former does not serialize the map keys properly. We should use go-yaml, which is the same used by clientconfig
> 
> Any examples?
> 
> more info on why k8s uses ghodss https://github.com/kubernetes-sigs/yaml

So, the config file generated by this would not serialize the keys as clientconfig expects them. The output is something like:

```
clouds:
    openstack:
        AuthUrl: # instead of auth_url
        Username: # instead of username
```

If there's something that one can set in ghodss to make it work as clientconfig expects, then I'm all for it. I may just lack of go-foo :smile:

**Comment by abhinavdahiya** [comment]
So looking at https://github.com/gophercloud/utils/blob/34f5991525d116b3832e0d9409492274f1c06bda/openstack/clientconfig/results.go

`ClientConfig` is only yaml serializable. `github.com/ghodss/yaml` uses the `json` tags to drive serialization to both json and yaml. 
So two options:
1. update `github.com/gophercloud/utils/openstack/clientconfig.ClientConfig` to include json tags...
2. add *second* import `gopkg.in/yaml.v2` just for `ClientConfig`

**Comment by flaper87** [comment]
@abhinavdahiya thanks a lot for the info. I'll work with the `gophercloud` team on adding the json tags for these structs.

Can we go with option 2 until the work on clientconfig is done? I'll put a task on me to clean this up as soon as the work in `clientconfig` lands

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya thanks a lot for the info. I'll work with the `gophercloud` team on adding the json tags for these structs.
> 
> Can we go with option 2 until the work on clientconfig is done? I'll put a task on me to clean this up as soon as the work in `clientconfig` lands

i'm fine with option 2 :innocent:

**Comment by flaper87** [comment]
cancelling the hold since @abhinavdahiya  is ok with option 2 while I work on moving `clientconfig` out of `go-yaml`

/hold cancel

**Comment by abhinavdahiya** [comment]
/hold

The go-yaml repo suggests `gopkg.in/yaml.v2` import . https://github.com/go-yaml/yaml/blob/v2/README.md#api-documentation

**Comment by flaper87** [comment]
@abhinavdahiya oh, totally missed that. Thanks!

/hold cancel

File: pkg/asset/manifests/openshift.go [hunk 1]
```diff
@@ -76,10 +83,15 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 			},
 		}
 	case "openstack":
-		clouds, err := clientconfig.LoadCloudsYAML()
+		opts := new(clientconfig.ClientOpts)
+		cloud, err := clientconfig.GetCloudFromYAML(opts)
```
Comment (hardys) [inline pkg/asset/manifests/openshift.go]: Do we also need to modify https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/openstack/openstack.go#L24 in a similar way?

Comment (flaper87) [reply pkg/asset/manifests/openshift.go]: No, this is only relevant for the marshal step. The install config step should work just fine.

Comment (hardys) [reply pkg/asset/manifests/openshift.go]: Ack thanks for clarifying

**Pull request merged**